README.md

```diff
@@ -548,6 +551,94 @@ Here's what they do:
+- for Blob transactions (EIP-4844) it's the sum of gas fee cap and tip cap and max fee per blob
+- for AccessList transactions (EIP-2930) it's just the gas price
+
+Please note that Blob and AccessList 
+
+If you want to use a custom bumping strategy, you can use a function with [GasBumpStrategyFn](retry.go) type. Here's an example of a custom strategy that bumps the gas price by 100% for every retry:
+```go
+var customGasBumpStrategyFn = func(gasPrice *big.Int) *big.Int {
+    return new(big.Int).Mul(gasPrice, big.NewInt(2))
+}
+```
+
+To use this strategy, you need to pass it to the `WithGasBumping` function in the `ClientBuilder`:
+```go
+var hundredGwei in64 = 100_000_000_000
+client, err := builder.
+    // other settings...
+    WithGasBumping(5, hundredGwei, customGasBumpStrategyFn).
+    Build()
+```
+
+Or set it directly on Seth's config:
+```go
+// assuming sethClient is already created
+sethClient.Config.GasBumps.StrategyFn = customGasBumpStrategyFn
+```
+
+Since strategy function only accepts a single parameter, if you want to base its behaviour on anything else than that you will need to capture these values from the context, in which you define the strategy function. For example, you can use a closure to capture the initial gas price:
+```go
+gasOracleClient := NewGasOracleClient()
+
+var oracleGasBumpStrategyFn = func(gasPrice *big.Int) *big.Int {
+    // get the current gas price from the oracle
+    suggestedGasPrice := gasOracleClient.GetCurrentGasPrice()
+
+	// if oracle suggests a higher gas price, use it
+    if suggestedGasPrice.Cmp(gasPrice) == 1 {
+        return suggestedGasPrice
+    }
+
+	// otherwise bump by 100%
+    return new(big.Int).Mul(gasPrice, big.NewInt(2))
+}
+```
+
+Currently, the gas bumping mechanism is available for:
+* Legacy transactions
+* EIP-1559 transactions.
+* Blob transactions (EIP-4844)
+* AccessList transactions (EIP-2930)
+
+Same strategy is applied to all types of transactions, regardless whether it's gas price, gas fee cap, gas tip cap or max blob fee.
+
+When enabled, gas bumping is used in two places:
+* during contract deployment via `DeployContract` function
+* inside `Decode()` function
+
+It is recommended to decrease transaction timeout when using gas bumping, as it will be effectively increased by the number of retries. So if you were running with 5 minutes timeout and 0 retries, you should set it to 1 minute and 5 retries
+or 30 seconds and 10 retries.
+
+Don't worry if while bumping logic executes previous transaction gets mined. In that case sending replacement transaction with higher gas will fail (because it is using the same nonce as original transaction) and we will retry waiting for the mining of the original transaction.
+
+**Gas bumping is only applied for submitted transaction. If transaction was rejected by the node (e.g. because of too low base fee) we will not bump the gas price nor try to submit it, because original transaction submission happens outside of Seth.**
```

Comment:
Sadly enough, that's true, haven't found any other possible workarounds?
`transact` in generated bindings returns `nil` if it fails to send
```
	if err := c.transactor.SendTransaction(ensureContext(opts.Context), signedTx); err != nil {
		return nil, err
	}
	return signedTx, nil
```

Reply:
yup so inside `Decode()` we don't have the transaction so we cannot resubmit it, the only solution I can think of is what I described to you yesterday: 
* implement `Backend`
* proxy `SendTransaction` to original one
* if err, retry inside our implementation of `SendTransaction`